src/main/java/uk/ac/ebi/biosamples/jsonschemastore/service/FieldRepositoryEventHandler.java

```diff
@@ -11,26 +12,76 @@
 import uk.ac.ebi.biosamples.jsonschemastore.model.FieldId;
 import uk.ac.ebi.biosamples.jsonschemastore.model.SchemaId;
 import uk.ac.ebi.biosamples.jsonschemastore.model.mongo.MongoJsonSchema;
+import uk.ac.ebi.biosamples.jsonschemastore.model.mongo.SchemaFieldAssociation;
+import uk.ac.ebi.biosamples.jsonschemastore.repository.FieldRepository;
+import uk.ac.ebi.biosamples.jsonschemastore.repository.SchemaRepository;
+
+import java.util.HashSet;
+import java.util.Set;
 
 import static uk.ac.ebi.biosamples.jsonschemastore.service.VariableNameFormatter.toVariableName;
 
 @Component
-@RepositoryEventHandler(Field.class)
+@RepositoryEventHandler
+@Slf4j
 public class FieldRepositoryEventHandler {
-    private static final Logger logger = LoggerFactory.getLogger(FieldRepositoryEventHandler.class);
-    @HandleBeforeCreate
-    public void handleBeforeCreate(Field field) {
-        field.setVersion("1.0");
-        field.setName(toVariableName(field.getLabel()));
-        field.setId(new FieldId(field.getName(), field.getVersion()).asString());
+  private final SchemaRepository schemaRepository;
+  private final FieldRepository fieldRepository;
+
+  public FieldRepositoryEventHandler(SchemaRepository schemaRepository, FieldRepository fieldRepository) {
+    this.schemaRepository = schemaRepository;
+    this.fieldRepository = fieldRepository;
+  }
+
+  @HandleBeforeCreate
+  public void handleBeforeCreate(Field field) {
+    field.setVersion("1.0");
+    field.setName(toVariableName(field.getLabel()));
+    field.setId(new FieldId(field.getName(), field.getVersion()).asString());
+  }
+
+  @HandleBeforeSave
+  public void handleBeforeSave(Field field) {
+    String oldFieldId = field.getId();
+    Field oldField = fieldRepository.findById(field.getId())
+        .orElseThrow(() -> new DataRetrievalFailureException("Could not find the field: " + oldFieldId));
+    if (!oldField.getLabel().equals(field.getLabel())) {
+      throw new DataIntegrityViolationException("Field `label` could not be edited once created. Please create a new field instead.");
```

Comment:
The term "Field" is ambiguous. The thing that cannot be changed is the attribute "label".
```suggestion
      throw new DataIntegrityViolationException("Attribute `label` could not be edited once created. Please create a new Field instead.");
```

Reply:
it was meant as label of the field, but if this is more friendly lets change it.